### PR TITLE
[cli] Support northstar users

### PR DESCRIPTION
**Pull request by chibicode**
Supports users who have `version === 'northstar'` and `defaultTeamId`.

### CODE REVIEW

File: .changeset/selfish-dancers-sort.md
Comment (EndangeredMassa) [inline .changeset/selfish-dancers-sort.md]: **suggestion (blocking):** I think we should call this a minor because it changes some behavior, but not technically in a backwards incompatible way (because you have to migrate your user account to encounter the new guards).‏

File: packages/cli/src/commands/login/index.ts
Comment (EndangeredMassa) [inline packages/cli/src/commands/login/index.ts]: **suggestion (blocking):** The exit code `2` should only be used for help output. Let's change this to `1`.

`‏
```suggestion
        return 1;
```

File: packages/cli/src/commands/teams/switch.ts
Comment (EndangeredMassa) [inline packages/cli/src/commands/teams/switch.ts]: **suggestion (non-blocking):** ‏extract into variable, or something

I find this complex set of expressions hard to read. Can we extract some named values or otherwise make this clearer?

File: packages/cli/src/util/errors-ts.ts
Comment (EndangeredMassa) [inline packages/cli/src/util/errors-ts.ts]: **suggestion (non-blocking):**

```suggestion
      message: `Your team was deleted or you were removed from the team. You can switch to a different one using ${getCommandName(
```

File: packages/cli/src/util/input/select-org.ts
Comment (EndangeredMassa) [inline packages/cli/src/util/input/select-org.ts]: **suggestion (non-blocking):** ‏extract into variable, or something

I find this complex set of expressions hard to read. Can we extract some named values or otherwise make this clearer?

**Review by EndangeredMassa — CHANGES_REQUESTED** [review]

File: .changeset/selfish-dancers-sort.md
Comment (chibicode) [inline .changeset/selfish-dancers-sort.md]:
```suggestion
"vercel": minor
"@vercel-internals/types": minor
```

File: packages/cli/src/commands/teams/switch.ts
Comment (chibicode) [reply packages/cli/src/commands/teams/switch.ts]: Fixed by 761ff070f

File: packages/cli/src/util/input/select-org.ts
Comment (chibicode) [reply packages/cli/src/util/input/select-org.ts]: Fixed by 761ff070f

**Review by EndangeredMassa — DISMISSED** [review]
Thanks! I think we should get one more review before merging.

File: .changeset/selfish-dancers-sort.md
Comment (styfle) [inline .changeset/selfish-dancers-sort.md]:
```suggestion
```

Private packages don't need to be versioned. You can remove this line

Comment (chibicode) [reply .changeset/selfish-dancers-sort.md]: @styfle thanks, fixed!

**Review by EndangeredMassa — DISMISSED** [review]

File: packages/cli/src/util/output/table.ts
Comment (styfle) [inline packages/cli/src/util/output/table.ts]: This is a breaking change since you are changing from printing `stdout` to `stderr`

File: packages/cli/test/integration-3.test.ts
Comment (styfle) [inline packages/cli/test/integration-3.test.ts]: This is a breaking change since you are changing from printing `stdout` to `stderr`

File: packages/cli/src/util/output/table.ts
Comment (chibicode) [reply packages/cli/src/util/output/table.ts]: @styfle fixed with e1096b84e

File: packages/cli/test/integration-3.test.ts
Comment (chibicode) [reply packages/cli/test/integration-3.test.ts]: @styfle fixed with e1096b84e

File: packages/cli/test/unit/commands/teams.test.ts
Comment (chibicode) [inline packages/cli/test/unit/commands/teams.test.ts]:
```suggestion
```

File: packages/cli/src/commands/teams/list.ts
Comment (chibicode) [inline packages/cli/src/commands/teams/list.ts:77]:
```suggestion
  output.print('\n'); // empty line
```

File: internals/types/index.d.ts
Comment (cb1kenobi) [inline internals/types/index.d.ts]: **suggestion (non-blocking):** ‏It's not clear to me what `version` is. It appears it will only ever be `'northstar'` or `undefined`, so maybe we just say that?

```suggestion
  version?: 'northstar';
```

File: packages/cli/src/commands/login/index.ts
Comment (cb1kenobi) [inline packages/cli/src/commands/login/index.ts]: Does this resolve https://linear.app/vercel/issue/ZERO-944/if-you-re-log-in-with-a-different-account-currentteam-will-be-out-of?

File: packages/cli/src/commands/login/index.ts
Comment (cb1kenobi) [inline packages/cli/src/commands/login/index.ts:77]: The "not a new login" comment above this is confusing given the test variable `isNewLogin`. The variable is probably fine, I think the comment just needs to say something like "If we have a new login, set the scope to the current team".

File: packages/cli/src/commands/teams/switch.ts
Comment (cb1kenobi) [inline packages/cli/src/commands/teams/switch.ts:77]: **question (non-blocking):** Will there be a "personal team" returned by `api` to replace this?‏

File: packages/cli/src/index.ts
Comment (cb1kenobi) [inline packages/cli/src/index.ts]: It would be nice to consolidate this logic with the virtually identical logic in `login.ts`. I wouldn't want one place to get changed and not all places.

File: packages/cli/src/index.ts
Comment (cb1kenobi) [inline packages/cli/src/index.ts]: **suggestion (non-blocking):** nitpick - The `else` isn't necessary, but I'm fine if you wanna leave it.‏

```suggestion
      }

      delete client.config.currentTeam;
```

File: packages/cli/src/commands/login/index.ts
Comment (styfle) [inline packages/cli/src/commands/login/index.ts]:
```suggestion
        output.error('Failed to fetch the logged in user. Please try again.');
```

**Review by styfle — DISMISSED** [review]
Looks good, thanks!

Might want to have @TooTallNate take a look since this changes auth/login

File: packages/cli/src/index.ts
Comment (chibicode) [inline packages/cli/src/index.ts]: Same as https://github.com/vercel/vercel/pull/10535#discussion_r1330333984

```suggestion
          console.error(error('Failed to fetch the logged in user. Please try again.'));
```

**Review by styfle — DISMISSED** [review]

File: packages/cli/src/commands/login/index.ts
Comment (chibicode) [reply packages/cli/src/commands/login/index.ts]: @cb1kenobi Unfortunately no, it'll require a separate fix.

File: packages/cli/src/commands/login/index.ts
Comment (chibicode) [inline packages/cli/src/commands/login/index.ts]: Per this comment: https://github.com/vercel/vercel/pull/10535/files/54434b23934e9053d56b11ae7f31c1feee1128c7#r1330176372

```suggestion
  // If we have a new login, update `currentTeam`
```

File: packages/cli/src/commands/login/index.ts
Comment (chibicode) [reply packages/cli/src/commands/login/index.ts:77]: Resolved by 07e7007bd

File: packages/cli/src/index.ts
Comment (chibicode) [reply packages/cli/src/index.ts]: @cb1kenobi Resolved by e579a3037

File: packages/cli/src/commands/teams/switch.ts
Comment (chibicode) [reply packages/cli/src/commands/teams/switch.ts:77]: Will communicate this via Slack!

File: packages/cli/src/commands/teams/list.ts
Comment (Ethan-Arrowood) [inline packages/cli/src/commands/teams/list.ts:77]: Hmm should this be `client.output` instead?

**Review by Ethan-Arrowood — DISMISSED** [review]
this is awesome. appreciate the very detailed tests

File: packages/cli/src/commands/teams/list.ts
Comment (Ethan-Arrowood) [reply packages/cli/src/commands/teams/list.ts:77]: Ignore me - found the discussion for this

**Review by EndangeredMassa — APPROVED** [review]

**Pull request merged**
